install.sh

```diff
@@ -180,6 +197,34 @@ show_summary() {
   fi
 }
 
+# =============================================================================
+# EARLY DEPENDENCY FUNCTIONS
+# =============================================================================
+
+# Install jq if needed for parameter file processing
+# Usage: ensure_jq_available
+# Returns: 0 if jq is available or successfully installed, 1 otherwise
+ensure_jq_available() {
+  # Install jq early if parameters file is provided and jq is not available
+  if [[ -n "$PARAMETERS_FILE" ]] && ! command_exists jq; then
+    info "jq is required for processing parameters file, installing early..."
+    
+    if ! command_exists brew; then
+      warn "Homebrew not available, cannot install jq. Parameter file processing will be skipped."
+      return 1
+    fi
```

Comment:
This will always happen because brew hasn't been installed yet because `cli_initial_setup.sh` hasn't been called yet. Maybe just call `cli_initial_setup.sh` before `create_links.sh` and that should take care of all these issues.